### PR TITLE
feat(authy): add filter and display features

### LINKED ISSUE #845: Hide disabled/deleted authy accounts
# Extension – [Authy](https://github.com/raycast/extensions/tree/44eb3fe084b7525eb28214738c18f7300c796a62/extensions/authy/)
Author: @guga4ka

**Describe the feature and the current behavior/state.**

The Authy app on iOS has the ability to hide `Authy Accounts` and mark `Authenticator Accounts` for deletion in 48 hours (these are called `Apps` and `Services` in the extension respectively). Currently both of these are still listed in the output from this extension. It would be nice if this extension could hide those as well. I haven't yet looked into whether these properties are returned by the API, so I'm not sure how viable this is.

Here's a screenshot of what it looks like in the iOS app accounts screen vs the extension. 

Screen Shot 2022-02-09 at 10 49 08 AM

**Who will benefit with this feature?**

I don't know of a way to delete `Authy Accounts` so anyone who has access to old ones that they no longer use will always see those in the output from this extension.

The hiding of `Authenticator Accounts` is less useful because they are permanently deleted in 48 hours, but for people who might be testing 2fa and making accounts this can be useful if they have many marked for deletion.

**Any Other info.**

I wanted to first see if this is a viable/wanted feature request, and if it is I would be happy to implement it. 😄

### PR DESCRIPTION
**Pull request by lukekarrys**
Hi @guga4ka! This PR will close the request I made in #845. I originally set out to just fork and maintain my own copy locally, but ended up deciding to make this PR. That's why there are a few additional features that I will outline below. It is currently all in one commit, but I'm happy to back out anything or make changes or open separate PRs.

Some of the features I added were also made easier by refactoring some of the code. Hopefully my changes aren't too subjective but in the case that it's harder to understand, I'm open to suggestions or I'm willing to take on `CODEOWNER`ship to help with future maintenance.

####  Adds `excludeNames` preference to optionally exclude accounts by name

This is the main feature I requested in #845. This implementation takes a preference as a comma separated list of values and removes any account matching one of those values. The comparison is done via `name.trim().toLocaleLowerCase()`.

#### Adds `separateAccounts` preference to optionally not separate accounts by type

I personally prefer to see all my accounts together regardless of type like the Authy native apps do. I added a preference for this behavior which defaults to `true` to match the current behavior, but depending on how others feel about it I think it might be better as the only possible behavior.

#### Adds support for looking up icons by the preferred `name` property as an additional fallback

Previously `logo` and `accountType` were looked up to find an icon match, and this adds `name` to the end of that list.

#### Hides item subtitle if it is the same as the name

This removes the `subtitle` if it would be redundant in addition to the `name`.

#### Refactors logos in `constants` for easier aliases

I noticed that `aws` was mapped to `amazonaws` already in the code, and I had a similar need to map `google apps` to `google`. So I update the data structure in `constants.js` to make this easier.

#### Adds some icons that I use personally

I realize this doesn't scale to have everyone add the icons they want, but I thought I'd try to sneak this in while working on this extension 😄.

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @guga4ka 

## Screencast

Here's a screenshot of the new preferences screen. The rest of the behavior is hard to show while maintaining my desired level of security hygiene 🤓.

Screen Shot 2022-02-09 at 5 39 53 PM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

closes #845

### CODE REVIEW

**Review by guga4ka — APPROVED** [review]
Awesome job @lukekarrys 🔥. Thx for your PR! 

Regarding the app's icon, I've tried to use favico for it, but I don't have the right url to get it. Currently, there isn't another way to add an icon, except a PR. In case you have ideas how to improve this process, I would be happy to do it.

File: extensions/authy/package.json [hunk 1]
```diff
@@ -47,6 +55,14 @@
...
       "description": "Primary action copies pin to clipboard instead of outputing to front application",
       "default": true
+    },
+    {
+      "required": false,
+      "name": "separateAccounts",
+      "label": "Separate apps and services",
+      "type": "checkbox",
+      "description": "Separate accounts in to apps and services",
+      "default": true
```
[the remark this answers is not in the document]
Comment (lukekarrys) [reply extensions/authy/package.json]: I removed this in the latest fixup commit.

**Comment by lukekarrys** [comment]
As for arbitrary icons at runtime, I don't have any ideas currently but might take some time to hack on this and see if I can come up with something.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @lukekarrys and @guga4ka 👋 

Awesome that youre working to great together making this extension even better 🔥 

Remember that you can add yourself as contributor by adding this to `package.json`

```
  "contributors": [
    "lukekarrys"
  ],
```

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Pull request merged**